src/dbnode/storage/index_block_test.go

```diff
@@ -657,6 +657,118 @@ func TestNamespaceIndexBlockQuery(t *testing.T) {
 	}
 }
 
+func TestLimits(t *testing.T) {
+	ctrl := xtest.NewController(t)
+	defer ctrl.Finish()
+
+	retention := 2 * time.Hour
+	blockSize := time.Hour
+	now := time.Now().Truncate(blockSize).Add(10 * time.Minute)
+	t0 := now.Truncate(blockSize)
+	t0Nanos := xtime.ToUnixNano(t0)
+	t1 := t0.Add(1 * blockSize)
+	var nowLock sync.Mutex
+	nowFn := func() time.Time {
+		nowLock.Lock()
+		defer nowLock.Unlock()
+		return now
+	}
+	opts := DefaultTestOptions()
+	opts = opts.SetClockOptions(opts.ClockOptions().SetNowFn(nowFn))
+
+	b0 := index.NewMockBlock(ctrl)
+	b0.EXPECT().Stats(gomock.Any()).Return(nil).AnyTimes()
+	b0.EXPECT().Close().Return(nil).AnyTimes()
+	b0.EXPECT().StartTime().Return(t0).AnyTimes()
+	b0.EXPECT().EndTime().Return(t0.Add(blockSize)).AnyTimes()
+	newBlockFn := func(
+		ts time.Time,
+		md namespace.Metadata,
+		_ index.BlockOptions,
+		io index.Options,
+	) (index.Block, error) {
+		if ts.Equal(t0) {
+			return b0, nil
+		}
+		panic("should never get here")
+	}
+	md := testNamespaceMetadata(blockSize, retention)
+	idx, err := newNamespaceIndexWithNewBlockFn(md, testShardSet, newBlockFn, opts)
+	require.NoError(t, err)
+
+	defer func() {
+		require.NoError(t, idx.Close())
+	}()
+
+	seg1 := segment.NewMockSegment(ctrl)
+	t0Results := result.NewIndexBlockByVolumeType(t0)
+	t0Results.SetBlock(idxpersist.DefaultIndexVolumeType, result.NewIndexBlock([]segment.Segment{seg1},
+		result.NewShardTimeRangesFromRange(t0, t1, 1, 2, 3)))
+	bootstrapResults := result.IndexResults{
+		t0Nanos: t0Results,
+	}
+
+	b0.EXPECT().AddResults(bootstrapResults[t0Nanos]).Return(nil)
+	require.NoError(t, idx.Bootstrap(bootstrapResults))
+
+	for _, test := range []struct {
+		name              string
+		seriesLimit       int
+		docsLimit         int
+		requireExhaustive bool
+		expectedErr       string
+	}{
+		{"no limits", 0, 0, false, ""},
```

Comment:
Can you make sure we use fields here (e.g. `{name: "no limits", seriesLimit: 0, ...}` etc). 

A lot of linters will complain about this, hence why we try to avoid doing this.

I believe our linter might actually be current broken (sadface...) hence why this doesn't cause errors. @schallert we should investigate why linter isn't complaining about this.